component-library/component-lib/src/lib/icon-button/icon-button.component.ts

```diff
@@ -1,18 +1,73 @@
-import { Component, OnInit } from '@angular/core';
+import {Component, Input, Output, EventEmitter, OnInit} from '@angular/core';
 export enum IconButtonCategories {
   primary = 'primary',
   critical = 'critical',
   custom = 'custom'
 }
+
+export enum IconButtonSize {
+  large = 'large',
+  small = 'small',
+  extraSmall = 'extraSmall'
+}
+
+export interface IconButtonIconConfig {
+  class: string; // Fontawesome icon class
+  color?: string; // icon color
+}
+
+export interface IconButtonComponentConfig {
+  id: string,
+  category: keyof typeof IconButtonCategories,
+  size?: keyof typeof IconButtonSize,
+  ariaLabel?: string,
+  disabled?: boolean,
+  customIcon?: IconButtonIconConfig
+}
 @Component({
   selector: 'lib-icon-button',
   templateUrl: './icon-button.component.html',
 })
 export class IconButtonComponent implements OnInit {
-
-  constructor() { }
+  @Input() config: IconButtonComponentConfig = {
+    id: '',
+    category: IconButtonCategories.primary,
+    ariaLabel: ''
+  };
+  @Input() id = '';
+  @Input()
+  category!: keyof typeof IconButtonCategories | IconButtonCategories;
+  @Input()
+  size!: keyof typeof IconButtonSize | IconButtonSize;
+  @Input()
+  ariaLabel?: string;
```

Comment:
Please fix the formatting so the @Input() and the declaration are on the same line.